.github/workflows/ci.yaml

```diff
@@ -0,0 +1,56 @@
+name: Test Suite
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: [3.8]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+    - name: Setup Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python }}
+    - name: Install dependencies
+      run: python3 -m pip install tox
+    - name: Run linters
+      run: tox -vve lint
+    - name: Static analysis
+      run: tox -vve static
+  integration-test-microk8s:
```

Comment:
I wonder if these could be common for our HA charms and part of shared workflows: https://github.com/canonical/observability/tree/main/.github/workflows